authservice/auth_service.go

```diff
@@ -271,47 +302,43 @@ func NewTestAuthService(serviceID string, serviceHost string, dataLoader AuthDat
 	return auth, nil
 }
 
-// -------------------- DataLoader --------------------
+// -------------------- ServiceAccountLoader --------------------
 
-// AuthDataLoader declares an interface to load data from an auth service
-type AuthDataLoader interface {
+// ServiceAccountLoader declares an interface to load service account data from an auth service
+type ServiceAccountLoader interface {
 	// GetAccessToken gets an access token
 	GetAccessToken() error
 	// GetDeletedAccounts loads deleted account IDs
 	GetDeletedAccounts() ([]string, error)
-	ServiceRegLoader
 }
 
-//RemoteAuthDataLoaderImpl provides a AuthDataLoader implementation for a remote auth service
-type RemoteAuthDataLoaderImpl struct {
-	config RemoteAuthDataLoaderConfig
+//RemoteServiceAccountLoaderImpl provides a ServiceAccountLoader implementation for a remote auth service
+type RemoteServiceAccountLoaderImpl struct {
+	config RemoteServiceAccountLoaderConfig
 
 	accessToken AccessToken
 
 	timerDone               chan bool
 	getDeletedAccountsTimer *time.Timer
 
 	logger *logs.Logger
-
-	*RemoteServiceRegLoaderImpl
 }
 
-//RemoteAuthDataLoaderConfig represents a configuration for a remote data loader
-type RemoteAuthDataLoaderConfig struct {
+//RemoteServiceAccountLoaderConfig represents a configuration for a remote service account loader
+type RemoteServiceAccountLoaderConfig struct {
 	AuthServicesHost string // URL of auth services host
 	ServiceToken     string // Static token issued by the auth service, used to get access tokens from the auth service
 
 	AccessTokenPath     string // Path to auth service access token endpoint
 	DeletedAccountsPath string // Path to auth service deleted accounts endpoint
```

Comment:
Deleted account endpoints are not used within the auth service so they should be moved to a separate `coreservice` package